wemake_python_styleguide/visitors/ast/builtins.py

```diff
@@ -274,6 +276,12 @@ def _check_unpacking_targets(
                     best_practices.WrongUnpackingViolation(node),
                 )
 
+    def _check_unpacking_target_type(self, node: ast.Assign) -> None:
+        if isinstance(node.targets[0], ast.List):
```

Comment:
Turns out we need to check all elements inside `targets` because of:

```python
>>> [a, b], c = ((1, 2), 3)
>>> a, [b, c] = (1, (2, 3))
```

Reply:
that's true. Should we stop detection on first spotted list or raise violations for all?

Reply:
unpacking to nested list actually raise `WPS414` - ` WrongUnpackingViolation`.
Should the second violation be raised?
 
I think we should do one of the following then:

+ update my PR to raise `WPS414` and update `WPS414` violations docstrings (I dislike this, because know `WPS414` can be raised in 2 totally different cases)
+ change name of `WPS414` to something less confusing and more compliant with its docstring title (`Forbids to have tuple unpacking with side-effects.`) and refactor check raising this violation, because unpacking to list doesn't have any side effects if it's not dynamic e.g `a, b[0] = (1, 3)` but `b[0]` is not parsed as `ast.List` by `ast`

What do you think?

Reply:
I think that these two violations should be independent. And both of them can be raised at the same time.